cmd/server/assets/apikeys/os.html

```diff
@@ -0,0 +1,76 @@
+{{$authApp := .authApp}}
+
+  <div class="form-group">
+    <select class="form-control{{if $authApp.ErrorsFor "os"}} is-invalid{{end}}" name="os" id="os" onchange="osSelect()">
+      <option value="0" {{if (eq $authApp.OS .unknownOS)}}selected{{else}}disabled{{end}}>Select OS...</option>
+      <option value="{{.iOS}}" {{if (eq $authApp.OS .iOS)}}selected{{end}}>iOS</option>
+      <option value="{{.android}}" {{if (eq $authApp.OS .android)}}selected{{end}}>Android</option>
+      <option value="{{.bothOS}}" {{if (eq $authApp.OS .bothOS)}}selected{{end}}>iOS and Android</option>
+    </select>
+    {{if $authApp.ErrorsFor "os"}}
+    <div class="invalid-feedback">
+      {{joinStrings ($authApp.ErrorsFor "os") ", "}}
+    </div>
+    {{end}}
+  </div>
+
+  <div class="form-label-group" id="iosDIV">
+    <input type="text" id="iosAppID" name="iosAppID" class="form-control{{if $authApp.ErrorsFor "iosAppID"}} is-invalid{{end}}" value="{{$authApp.IOSAppID}}" placeholder="Application ID" autofocus>
+    <label for="iosAppID">iOS Application ID</label>
+    {{if $authApp.ErrorsFor "iosAppID"}}
+    <div class="invalid-feedback">
+      {{joinStrings ($authApp.ErrorsFor "iosAppID") ", "}}
+    </div>
+    {{end}}
+  </div>
+
+  <div class="form-label-group" id="androidDIV">
+    <input type="text" id="androidAppID" name="androidAppID" class="form-control{{if $authApp.ErrorsFor "androidAppID"}} is-invalid{{end}}" value="{{$authApp.AndroidAppID}}" placeholder="Application ID" autofocus>
+    <label for="androidAppID">Android Application ID</label>
+    {{if $authApp.ErrorsFor "androidAppID"}}
+    <div class="invalid-feedback">
+      {{joinStrings ($authApp.ErrorsFor "androidAppID") ", "}}
+    </div>
+    {{end}}
+  </div>
+
+  <div class="form-label-group" id="shaDIV">
+    <input type="text" id="sha" name="sha" class="form-control{{if $authApp.ErrorsFor "sha"}} is-invalid{{end}}" value="{{$authApp.SHA}}" placeholder="SHA" autofocus>
+    <label for="sha">SHA</label>
+    {{if $authApp.ErrorsFor "sha"}}
+    <div class="invalid-feedback">
+      {{joinStrings ($authApp.ErrorsFor "sha") ", "}}
+    </div>
+    {{end}}
+  </div>
```

Comment:
I can see in other files that the underlying storage & code support multiple SHA hashes per app, which is great. I'm not familiar with this html templating engine, so I can't tell if the UI intends to support multiple hashes "top to bottom". If so great, but if not, it might be good to do so, to help smooth integration with the app, given developers will need to test with a debug-keyed build prior to publishing a release-keyed build.

Reply:
Do we require multiple SHA hashes per app? Currently, I don't support this, but could.

Reply:
Moved to supporting multiple SHAs for an android app.